src/AppInstallerCLICore/Workflows/DependenciesFlow.cpp

```diff
@@ -133,6 +135,109 @@ namespace AppInstaller::CLI::Workflow
+                    const auto& name = dependency.Id();
+                    WindowsFeature::WindowsFeature windowsFeature{ name };
+                    if (!windowsFeature.DoesExist())
+                    {
+                        invalidFeatures.emplace_back(name);
+                    }
+                });
+
+            if (!invalidFeatures.empty())
+            {
+                bool shouldTerminate = true;
+                auto warn = context.Reporter.Warn();
+                if (context.Args.Contains(Execution::Args::Type::IgnoreMissingDependencies))
+                {
+                    warn << Resource::String::WindowsFeatureNotFoundOverride << std::endl;
+                    shouldTerminate = false;
+                }
+                else
+                {
+                    warn << Resource::String::WindowsFeatureNotFoundOverrideRequired << std::endl;
+                }
+
+                warn << "  - " << Resource::String::WindowsFeaturesDependencies << std::endl;
+
+                for (const auto& feature : invalidFeatures)
+                {
+                    warn << "      " << feature << std::endl;
+                }
+
+                if (shouldTerminate)
+                {
+                    AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_INSTALL_MISSING_DEPENDENCY);
+                }
+            }
+
+            HRESULT hr = S_OK;
+            bool continueOnFailure = context.Args.Contains(Execution::Args::Type::Force);
+            bool rebootRequired = false;
+
+            context.Reporter.Info() << Resource::String::EnablingWindowsFeatures << std::endl;
+            rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&hr, &continueOnFailure, &rebootRequired](Dependency dependency)
+                {
+                    if (SUCCEEDED(hr) || continueOnFailure)
+                    {
+                        auto featureName = dependency.Id();
+                        WindowsFeature::WindowsFeature windowsFeature{ featureName };
+                        if (windowsFeature.DoesExist() && !windowsFeature.IsEnabled())
+                        {
+                            hr = windowsFeature.EnableFeature();
+                            AICLI_LOG(Core, Info, << "Enabling Windows Feature " << featureName << " returned with HRESULT: " << hr);
+                            if (hr == ERROR_SUCCESS_REBOOT_REQUIRED)
+                            {
+                                rebootRequired = true;
+                            }
+                        }
+                    }
+            });
+
+            if (FAILED(hr))
+            {
+                if (continueOnFailure)
+                {
+                    context.Reporter.Warn() << Resource::String::FailedToEnableWindowsFeatureOverridden << std::endl;
+                }
+                else
+                {
+                    context.Reporter.Error() << Resource::String::FailedToEnableWindowsFeatureOverrideRequired << std::endl;

```

Comment:
@denelon - I may be asking the wrong (scope creep) questions here, so forgive me, but would we want any sort of telemetry around these?
1) Which features are requested to be enabled
2) The success / fail rates of enabling features

Reply:
I think we should be showing each feature as its own item and not aggregating them as well.

Reply:
Changed so that each windows feature is represented as its own item.

Reply:
Sounds good; I'm presuming any telemetry needs can be added once support is no longer experimental